### PR TITLE
Add Python binomial test

### PR DESCRIPTION
**Pull request by thisisnic**
Draft PR to add Python binomial test - I think we'll need to update requirements.txt as well.  

We could also add guidance around how to update requirements.txt.

### CODE REVIEW

**Comment by LucaAnholt** [comment]
Looks like the build failure might be down to a `scipy` version issue. 

In the [requirments.txt](https://github.com/PSIAIMS/CAMIS/blob/main/requirements.txt#L85C1-L85C14) scipy is at version `1.13.0`, and from looking online it looks like the `binom_test` module was [refactored](https://stackoverflow.com/a/78090496) in version `1.12.0`.

File: python/binomial_test.qmd [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+## Creating a sample dataset
+
+-   We will generate a dataset where we record the outcomes of 1000 coin flips.
+
+-   We will use the `binom.test` function to test if the proportion of heads is significantly different from 0.5.
+
+```{python}
+import numpy as np
+from scipy.stats import binom_test
```
Comment (thisisnic) [inline python/binomial_test.qmd:17]:
```suggestion
from scipy.stats import binomtest
```

File: python/binomial_test.qmd [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+total_flips = len(coin_flips)
+
+heads_count, tails_count, total_flips
+```
+
+## Conducting Binomial Test
+
+```{python}
+# Perform the binomial test
+binom_test_result = binom_test(heads_count, total_flips, p=0.5)
```
Comment (thisisnic) [inline python/binomial_test.qmd:39]:
```suggestion
binom_test_result = binomtest(heads_count, total_flips, p=0.5)
```

File: python/binomial_test.qmd [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+
+
+## Conduct the Binomial Test
+
+We will conduct the Binomial test and hypothesize that the proportion of death should be 19%.
+
+```{python}
+# Perform the binomial test
+binom_test_clinical = binom_test(num_deaths, total_pat, p=0.19)
```
Comment (thisisnic) [inline python/binomial_test.qmd:75]:
```suggestion
binom_test_clinical = binomtest(num_deaths, total_pat, p=0.19)
```

File: python/binomial_test.qmd [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+```{python}
+# Perform the binomial test
+binom_test_result = binomtest(heads_count, total_flips, p=0.5)
+binom_test_result
+```
+
+### Results:
+
+The output has a p-value `r binom_test_result$p.value` $> 0.05$ (chosen level of significance). Hence, we fail to reject the null hypothesis and conclude that the **coin is fair**.
```
Comment (thisisnic) [inline python/binomial_test.qmd:45]:
```suggestion
The output has a p-value `py binom_test_result` $> 0.05$ (chosen level of significance). Hence, we fail to reject the null hypothesis and conclude that the **coin is fair**.
```

**Comment by thisisnic** [comment]
In the end I removed the additional dependency which was needed for the dataset and just used the CSV file instead!

**Review by orladoylenvs — APPROVED** [review]
LGTM!

**Pull request merged**